templates/test/infopanel.html.ep

```diff
@@ -96,5 +96,10 @@
                         </span>
                     </div>
                 % }
+                % if ($job && $job->scenario && $job->scenario->description && length($job->scenario->description)) {
```

Comment:
I would avoid that many method calls just for getting the same value again and again. According to @kraih it is not the best practice in Perl performance-wise. Also the check for `length` is useless because an empty string evaluates to `false` in Perl.

So I'd rather add this line without the multiple method calls and length check to the controller and stash the scenario description there.

Reply:
I just saw that we already have a convenience function for the description. So just write it like this:

```
if (my $scenario_description = $job ? $job->scenario_description : undef) {
```

Reply:
The thing I initially didn't like and still think should be improved was the unnecessarily repeated method call. It should be `if (my $scenario_description = $job->scenario_description) {` to use the variable inside the block. (That saves more than one method call because `$job->scenario_description` does further method calls internally.)